### ISSUE TITLE
RenderTexture.Create failed: requested size is too large.

### ISSUE THREAD

**Issue by giggioz** [jeffreylanters/react-unity-webgl#254]
Hi,

I've just created a Unity WebGL (Unity Version 2020.3.20f1) build with a cube object.

Relevant settings

![image](https://user-images.githubusercontent.com/6266740/140932592-9ce93182-e134-4179-a5f1-d48bc7f80382.png)

![image](https://user-images.githubusercontent.com/6266740/140932514-87c74b85-2e4d-4c3f-8027-173664351cc2.png)

Then I created a react app with create-react-app, I imported react-unity-webgl 8.6.0

and changed the App.js file:

```
import logo from './logo.svg';
import './App.css';
import Unity, { UnityContext } from 'react-unity-webgl';

const unityContext = new UnityContext({
  loaderUrl: 'unity-app/Build/unity-app.loader.js',
  dataUrl: 'unity-app/Build/unity-app.data.unityweb',
  frameworkUrl: 'unity-app/Build/unity-app.framework.js.unityweb',
  codeUrl: 'unity-app/Build/unity-app.wasm.unityweb',
});

function App() {
  return (
    <div className='App'>
      <header className='App-header'>
        <img src={logo} className='App-logo' alt='logo' />
        <div>
          <p>UNITY PLAYER HERE</p>
          <Unity style={{border:'1px solid red'}} unityContext={unityContext} />
        </div>
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className='App-link'
          href='https://reactjs.org'
          target='_blank'
          rel='noopener noreferrer'
        >
          Learn React
        </a>
      </header>
    </div>
  );
}

export default App;
```

![image](https://user-images.githubusercontent.com/6266740/140934363-066cb411-a90d-4b46-aef2-8549b694c6cb.png)

running the app I get a black div

![image](https://user-images.githubusercontent.com/6266740/140932784-7c125028-204b-42bc-86e9-ab498213f3a0.png)

and this looping error

![image](https://user-images.githubusercontent.com/6266740/140932825-1eb07b96-0eea-4806-abd7-8efbfbf1b76c.png)

![image](https://user-images.githubusercontent.com/6266740/140932860-d97c6395-a09e-4263-adf3-5ff899d616cc.png)

Just an extra info: I tried also with a more complex Unity Project, same issue but I can listen the music I put in the game... so the player is somehow loaded...

Extra info 2: If I build and run from Unity without the React project the WebGL app works

![image](https://user-images.githubusercontent.com/6266740/140936370-2f800878-e7dd-4e71-9d04-e6fe48d5de91.png)

Can you help me on this?
Thanks a lot

**Comment by giggioz** [comment]
Wow, after some trial and error I found a simple solution:

just use **maxWidth** in the style of the component

I leave the issue open if you want to dig into it, thanks!

**Comment by jeffreylanters** [comment]
Hi. I'm glad you've found a solution and thanks for your detailed report.

To clarify with actually happens; the height and width provided in Unity have nothing to do with the actual build. All Unity does is place these number in the generated HTML file. But since we're not using this file but only focus on the actual builded JS and Binary files, this data is no longer relevant. 

Applying no height and width to the Unity component results in the browser falling back on the default Canvas element height and width. In some browsers this doesn't work the way it should in combination with Unity, and the canvas scales indefinitely, resulting in these specific errors. 

TLDR; Providing a simple height and width to the Unity component solves these issues.

Happy coding!
